src/colorant.rs

```diff
@@ -6,10 +6,28 @@ use nalgebra::SVector;
 use crate::{physics::{gaussian_peak_one, wavelength}, spectrum::wavelengths, error::CmtError, traits::Filter, spectrum::{Spectrum, NS}};
 
 
+/// A Colorant represents color filters and color patches. These are spectrums
+/// with spectral values between 0.0 and 1.0.
 #[derive(Clone, Debug, Default, PartialEq)]
 pub struct Colorant(pub(crate) Spectrum);
 
 impl Colorant {
+    /// Creates a Colorant from a spectrum, while validating the spectrum values.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the spectrum contains values outside the range 0.0 to 1.0.
```

Comment:
It is best practice ([according to the Rust API guidelines](https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure)) to document any failures like these under an explicit `# Errors` header in the docs.